src/app/volunteer/register/[...slug]/page.tsx

```diff
@@ -0,0 +1,121 @@
+'use client';
+import FormProvider from '@/components/common/FormProvider/FormProvider';
+import useFunnel from '@/hooks/useFunnel';
+import { useForm } from 'react-hook-form';
+import NickName from './NickName';
+import Button from '@/components/common/Button/Button';
+import * as style from './style.css';
+import { useSetRecoilState } from 'recoil';
+import { headerState } from '@/store/header';
+import { useEffect, useMemo } from 'react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import ContactNumber from './ContactNumber';
+import {
+  VolunteerRegisterPayload,
+  postVolunteerRegister
+} from '@/api/auth/register';
+import { fetchCheckNickname } from '@/api/user/nickname';
+import { validation } from './validationSchema';
+import RegisterComplete from '@/components/volunteer/RegisterComplete/RegisterComplete';
+
+export type RegisterFormValues = {
+  nickName: string;
+  contactNumber: string;
+  complete: string;
+};
+type RegisterSteps<T> = {
+  component: () => JSX.Element;
+  path: keyof T;
+  asyncCheck: Function;
+};
+
+export default function RegisterPage({ params }: { params: { slug: string } }) {
+  const Steps: RegisterSteps<RegisterFormValues>[] = useMemo(
+    () => [
+      {
+        component: NickName,
+        path: 'nickName',
+        asyncCheck: async () => {
+          const query = getValues('nickName');
+          const check = await fetchCheckNickname(query || '');
+          if (check === true) throw new Error('이미 존재하는 닉네임입니다.');
+        }
+      },
+      {
+        component: ContactNumber,
+        path: 'contactNumber',
+        asyncCheck: async () => {
+          const payload: VolunteerRegisterPayload = {
+            nickname: getValues('nickName') || '',
+            phone: getValues('contactNumber') || '',
+            email: 'sangjun@naver.com' as string // 이메일을 알수가엇ㅂ어..
+          };
+          return await postVolunteerRegister(payload);
+        }
+      },
+      {
+        component: RegisterComplete,
+        path: 'complete',
+        asyncCheck: async () => {}
+      }
+    ],
+    []
+  );
+
+  const pathname = `/volunteer/register`;
+  const setHeader = useSetRecoilState(headerState);
```

Comment:
종이님께서 useHeader 훅을 만들어주셨어요!
```suggestion
  useHeader({ title: '기본 설정' });
```